lib/service/database.js

```diff
@@ -66,28 +66,42 @@ var affectedTableRegexCache = {
     pixelHeight: /!pixel_height!/g
 };
 
-DatabaseService.prototype.getLastUpdatedTimeFromAffectedTables = function(node, skip, callback) {
+DatabaseService.prototype.getLastUpdatedTimeAndTableDataFromAffectedTables = function(node, skip, callback) {
     if (!!skip || node.type !== Source.TYPE || node.getUpdatedAt() !== null) {
-        return callback(null, node.getUpdatedAt());
+        var data = {'last_update': node.getUpdatedAt(),
+                    'affected_tables': node.getAffectedTables()};
+        return callback(null, data);
     }
 
     var sql = node.sql()
         .replace(affectedTableRegexCache.bbox, 'ST_MakeEnvelope(0,0,0,0)')
         .replace(affectedTableRegexCache.scaleDenominator, '0')
         .replace(affectedTableRegexCache.pixelWidth, '1')
         .replace(affectedTableRegexCache.pixelHeight, '1');
-    sql = 'SELECT max(updated_at) FROM CDB_QueryTables_Updated_At($camshaft$' + sql + '$camshaft$)';
+    sql = 'SELECT updated_at, schema_name, table_name ' +
+          'FROM CDB_QueryTables_Updated_At($camshaft$' + sql + '$camshaft$)';
 
     this.queryRunner.run(sql, QUERY_RUNNER_READONLY_OP, function(err, resultSet) {
         if (err) {
             return callback(err);
         }
 
         var defaultLastUpdatedTime = new Date('1970-01-01T00:00:00.000Z');
-        var rows = resultSet.rows || [defaultLastUpdatedTime];
-        var lastUpdatedTimeFromAffectedTables = (rows[0] && rows[0].max) || defaultLastUpdatedTime;
+        var affectedTables = [];
+        var lastUpdatedTimes = [];
+        var rows = resultSet.rows || [defaultLastUpdatedTime, null, null];
```

Comment:
Yes because we still want to store the last updated time. I didn't want to change anything, at last null values in the affected tables are an empty array